tools/python/boutiques/publisher.py

```diff
@@ -169,6 +125,21 @@ def zenodo_deposit(self):
                                    format(zid), r)
         return zid
 
+    def zenodo_deposit_updated_version(self, deposition_id):
+        r = requests.post(self.zenodo_endpoint +
+                          '/api/deposit/depositions/%s/actions/newversion'
+                          % deposition_id,
+                          params={'access_token': self.zenodo_access_token})
+        if(r.status_code != 201):
+            self.raise_zenodo_error("Deposition of new version failed", r)
+        if(self.verbose):
+            self.print_zenodo_info("Deposition of new version succeeded", r)
+        new_url = r.json()['links']['latest_draft']
+        new_zid = new_url.split("/")[-1]
+        self.zenodo_update_metadata(new_zid)
+        self.zenodo_delete_files(new_zid, r.json()["files"])
```

Comment:
what are you deleting here?

Reply:
When you create a new version, Zenodo copies everything from the previous version, including the uploaded files. So here I am deleting any files from the previous version before uploading the updated file. If you don't delete the files and you try to upload one with the same name, their server throws a 400 error. If you don't delete and you try to upload one with a different name, the new version will have both the old and the new version, which I would assume the user does not want.